modules/users/server/models/user.server.model.js

```diff
@@ -112,7 +112,7 @@ UserSchema.pre('save', function(next) {
  */
 UserSchema.methods.hashPassword = function(password) {
 	if (this.salt && password) {
-		return crypto.pbkdf2Sync(password, this.salt, 10000, 64).toString('base64');
+		return crypto.pbkdf2Sync(password, new Buffer(this.salt, 'base64'), 10000, 64).toString('base64');
```

Comment:
The code could be:
```js
UserSchema.methods.hashPassword = function(password, saltBuffer) {
  if (saltBuffer) {
    return crypto.pbkdf2Sync(password, saltBuffer, 1000, 64).toString('base64');
  } else if (this.salt) {
     return crypto.pbkdf2Sync(password, new Buffer(this.salt, 'base64'), 1000, 64).toString('base64');
  } else {
    return password;
  }
};
```

Wouldn't not re-instantiating be much more performant for long passwords? Can you set up some sort of JSPerf or benchmark to figure out whether we should do it or not?